onnx/__init__.py

```diff
@@ -12,29 +12,31 @@
 import onnx.defs  # noqa
 
 import google.protobuf.message  # type: ignore
+from typing import Union, Optional, Text, IO
 
+from typing import Union, Text, IO, Optional, cast, TypeVar, Any
 
 # f should be either readable or a file path
-def _load_bytes(f):
-    if hasattr(f, 'read') and callable(f.read):
-        s = f.read()
+def _load_bytes(f):  # type: (Union[IO[bytes], Text]) -> bytes
+    if hasattr(f, 'read') and callable(f.read):  # type: ignore
```

Comment:
why do we need to mark ignore here?

Reply:
because mypy doesn't support hasattr() checks yet and complains about the `f.read` (since Text doesn't have a .read member). But I removed it now and replaced it with a type cast, which is a bit less intrusive.